Index: src/gtk+-3.24.12/gtk/gtkwindow.c
===================================================================
--- src.orig/gtk+-3.24.12/gtk/gtkwindow.c
+++ src/gtk+-3.24.12/gtk/gtkwindow.c
@@ -8004,7 +8004,7 @@ gtk_window_state_event (GtkWidget
   GtkWindow *window = GTK_WINDOW (widget);
   GtkWindowPrivate *priv = window->priv;
 
-  if (event->changed_mask & GDK_WINDOW_STATE_FOCUSED)
+  if ((event->changed_mask & GDK_WINDOW_STATE_FOCUSED) && priv->client_decorated)
     ensure_state_flag_backdrop (widget);
 
   if (event->changed_mask & GDK_WINDOW_STATE_FULLSCREEN)
