diff --color -U 5 -r -Z -B ./org/gtk/gtkwindow.c ./mod/gtk/gtkwindow.c
--- ./org/gtk/gtkwindow.c	2017-10-27 22:06:52.000000000 +0200
+++ ./mod/gtk/gtkwindow.c	2017-10-30 21:15:37.183638087 +0100
@@ -4087,10 +4087,11 @@
 }
 
 static gboolean
 gtk_window_supports_client_shadow (GtkWindow *window)
 {
+  if (g_strcmp0 (g_getenv("GTK_CSD"), "1") != 0) return FALSE;
   GdkDisplay *display;
   GdkScreen *screen;
   GdkVisual *visual;
 
   screen = _gtk_window_get_screen (window);
