diff --color -U 5 -r -Z -B ./org/gtk/gtkwindow.c ./mod/gtk/gtkwindow.c
--- ./org/gtk/gtkwindow.c	2017-10-27 22:06:52.000000000 +0200
+++ ./mod/gtk/gtkwindow.c	2017-10-30 21:04:21.086937922 +0100
@@ -4147,11 +4147,11 @@
   else
     {
       gtk_style_context_add_class (gtk_widget_get_style_context (widget), "solid-csd");
     }
 
-  priv->client_decorated = TRUE;
+  priv->client_decorated = (g_getenv ("GTK_CSD") != FALSE);
 }
 
 static void
 on_titlebar_title_notify (GtkHeaderBar *titlebar,
                           GParamSpec   *pspec,
