diff --color -U 5 -r -Z -B ./org/gtk/gtkwindow.c ./mod/gtk/gtkwindow.c
--- ./org/gtk/gtkwindow.c	2017-10-27 22:06:52.000000000 +0200
+++ ./mod/gtk/gtkwindow.c	2017-10-30 21:27:43.370340735 +0100
@@ -7428,10 +7428,12 @@
     }

   gtk_widget_set_window (widget, gdk_window);
   gtk_widget_register_window (widget, gdk_window);
   gtk_widget_set_realized (widget, TRUE);
+  if (!gdk_screen_is_composited (priv->screen) && !gtk_widget_get_app_paintable (widget))
+    gtk_style_context_set_background (gtk_widget_get_style_context (widget), gdk_window);

   attributes.x = allocation.x;
   attributes.y = allocation.y;
   attributes.width = allocation.width;
   attributes.height = allocation.height;
