diff --color -U 5 -r -Z -B ./org/gtk/gtkwindow.c ./mod/gtk/gtkwindow.c
--- ./org/gtk/gtkwindow.c	2018-01-15 14:21:38.000000000 +0100
+++ ./mod/gtk/gtkwindow.c	2018-06-01 23:18:47.828882837 +0200
@@ -7435,10 +7435,15 @@
     }
 
   gtk_widget_set_window (widget, gdk_window);
   gtk_widget_register_window (widget, gdk_window);
   gtk_widget_set_realized (widget, TRUE);
+#ifdef GDK_WINDOWING_X11
+  if (GDK_IS_X11_DISPLAY (gtk_widget_get_display (widget)) && !gtk_widget_get_app_paintable (widget) &&
+      !gdk_x11_screen_supports_net_wm_hint (priv->screen, gdk_atom_intern_static_string ("_GTK_FRAME_EXTENTS")))
+    gtk_style_context_set_background (gtk_widget_get_style_context (widget), gdk_window);
+#endif
 
   attributes.x = allocation.x;
   attributes.y = allocation.y;
   attributes.width = allocation.width;
   attributes.height = allocation.height;
