diff --color -U 5 -r -Z -B ./org/gdk/x11/gdkwindow-x11.c ./mod/gdk/x11/gdkwindow-x11.c
--- ./org/gdk/x11/gdkwindow-x11.c	2018-06-19 17:45:21.000000000 +0200
+++ ./mod/gdk/x11/gdkwindow-x11.c	2018-09-21 23:44:50.443842645 +0200
@@ -2983,10 +2983,20 @@
   if (GDK_WINDOW_DESTROYED (window))
     return;
 
   if (pattern == NULL)
     {
+      gpointer gtk_widget_ptr;
+      GObject *gtk_widget;
+      gdk_window_get_user_data (window, &gtk_widget_ptr);
+      gtk_widget = G_OBJECT (gtk_widget_ptr);
+      if (gtk_widget && g_strcmp0 (G_OBJECT_TYPE_NAME (gtk_widget), "GtkTrayIcon") == 0)
+      {
+        XSetWindowBackgroundPixmap (GDK_WINDOW_XDISPLAY (window),
+                                    GDK_WINDOW_XID (window), ParentRelative);
+        return;
+      }
       XSetWindowBackgroundPixmap (GDK_WINDOW_XDISPLAY (window),
                                   GDK_WINDOW_XID (window), None);
       return;
     }
 
