diff --color -U 5 -r -Z -B ./org/gtk/gtkapplication-dbus.c ./mod/gtk/gtkapplication-dbus.c
--- ./org/gtk/gtkapplication-dbus.c	2018-09-14 05:32:45.000000000 +0200
+++ ./mod/gtk/gtkapplication-dbus.c	2018-09-22 00:05:37.731044507 +0200
@@ -644,11 +644,11 @@
           return cookie;
         }
 
       if (error)
         {
-          if (!warned)
+          if (FALSE)
             {
               g_warning ("Calling %s.Inhibit failed: %s",
                          g_dbus_proxy_get_interface_name (dbus->sm_proxy),
                          error->message);
               warned = TRUE;
