diff -U 5 -r -Z -B ./org/gtk/gtkscrolledwindow.c ./mod/gtk/gtkscrolledwindow.c
--- ./org/gtk/gtkscrolledwindow.c	2017-05-10 20:57:00.000000000 +0200
+++ ./mod/gtk/gtkscrolledwindow.c	2017-05-11 19:11:54.443652914 +0200
@@ -4398,11 +4398,11 @@
   GtkScrolledWindowPrivate *priv = scrolled_window->priv;
   gboolean use_indicators;
 
   use_indicators = priv->overlay_scrolling;
 
-  if (g_strcmp0 (g_getenv ("GTK_OVERLAY_SCROLLING"), "0") == 0)
+  if (TRUE || g_strcmp0 (g_getenv ("GTK_OVERLAY_SCROLLING"), "0") == 0)
     use_indicators = FALSE;
 
   if (priv->use_indicators != use_indicators)
     {
       priv->use_indicators = use_indicators;
diff -U 5 -r -Z -B ./org/gtk/gtksettings.c ./mod/gtk/gtksettings.c
--- ./org/gtk/gtksettings.c	2017-05-10 20:57:00.000000000 +0200
+++ ./mod/gtk/gtksettings.c	2017-05-11 19:12:01.523653266 +0200
@@ -1721,11 +1721,11 @@
    */
   result = settings_install_property_parser (class,
                                              g_param_spec_boolean ("gtk-recent-files-enabled",
                                                                    P_("Recent Files Enabled"),
                                                                    P_("Whether GTK+ remembers recent files"),
-                                                                   TRUE,
+                                                                   FALSE,
                                                                    GTK_PARAM_READWRITE),
                                              NULL);
   g_assert (result == PROP_RECENT_FILES_ENABLED);
 
   /**
diff -U 5 -r -Z -B ./org/gtk/gtkwindow.c ./mod/gtk/gtkwindow.c
--- ./org/gtk/gtkwindow.c	2017-05-10 20:57:00.000000000 +0200
+++ ./mod/gtk/gtkwindow.c	2017-05-11 19:12:26.660321178 +0200
@@ -149,11 +149,11 @@
  *
  * GtkWindow adds the .titlebar and .default-decoration style classes to the
  * widget that is added as a titlebar child.
  */
 
-#define MNEMONICS_DELAY 300 /* ms */
+#define MNEMONICS_DELAY 0 /* ms */
 #define NO_CONTENT_CHILD_NAT 200
 /* In case the content (excluding header bar and shadows) of the window
  * would be empty, either because there is no visible child widget or only an
  * empty container widget, we use NO_CONTENT_CHILD_NAT as natural width/height
  * instead.
diff -U 5 -r -Z -B ./org/gtk/ui/gtkstatusbar.ui ./mod/gtk/ui/gtkstatusbar.ui
--- ./org/gtk/ui/gtkstatusbar.ui	2017-05-10 20:57:00.000000000 +0200
+++ ./mod/gtk/ui/gtkstatusbar.ui	2017-05-11 21:02:41.253982250 +0200
@@ -3,12 +3,12 @@
   <!-- interface-requires gtk+ 3.10 -->
   <template class="GtkStatusbar" parent="GtkBox">
     <property name="spacing">2</property>
     <property name="margin-start">10</property>
     <property name="margin-end">10</property>
-    <property name="margin-top">6</property>
-    <property name="margin-bottom">6</property>
+    <property name="margin-top">1</property>
+    <property name="margin-bottom">1</property>
     <child>
       <object class="GtkFrame" id="frame">
         <property name="visible">1</property>
         <property name="shadow-type">none</property>
         <child>
