diff -U 5 -r -Z -B ./org/gtk/gtkfilesystem.c ./mod/gtk/gtkfilesystem.c
--- ./org/gtk/gtkfilesystem.c	2017-07-19 04:55:39.000000000 +0200
+++ ./mod/gtk/gtkfilesystem.c	2017-07-30 13:48:51.953531745 +0200
@@ -775,17 +775,17 @@
 
 GIcon *
 _gtk_file_system_volume_get_symbolic_icon (GtkFileSystemVolume *volume)
 {
   if (IS_ROOT_VOLUME (volume))
-    return g_themed_icon_new ("drive-harddisk-symbolic");
+    return g_themed_icon_new ("drive-harddisk");
   else if (G_IS_DRIVE (volume))
-    return g_drive_get_symbolic_icon (G_DRIVE (volume));
+    return g_drive_get_icon (G_DRIVE (volume));
   else if (G_IS_VOLUME (volume))
-    return g_volume_get_symbolic_icon (G_VOLUME (volume));
+    return g_volume_get_icon (G_VOLUME (volume));
   else if (G_IS_MOUNT (volume))
-    return g_mount_get_symbolic_icon (G_MOUNT (volume));
+    return g_mount_get_icon (G_MOUNT (volume));
   else
     return NULL;
 }
 
 GtkFileSystemVolume *
diff -U 5 -r -Z -B ./org/gtk/gtkpathbar.c ./mod/gtk/gtkpathbar.c
--- ./org/gtk/gtkpathbar.c	2017-07-19 04:55:39.000000000 +0200
+++ ./mod/gtk/gtkpathbar.c	2017-07-30 13:48:51.956865078 +0200
@@ -1271,11 +1271,11 @@
     }
 
   if (cancelled || error)
     goto out;
 
-  icon = g_file_info_get_symbolic_icon (info);
+  icon = g_file_info_get_icon (info);
   gtk_image_set_from_gicon (GTK_IMAGE (data->button_data->image), icon, GTK_ICON_SIZE_BUTTON);
 
   switch (data->button_data->type)
     {
       case HOME_BUTTON:
@@ -1337,11 +1337,11 @@
 	g_cancellable_cancel (button_data->cancellable);
 
       button_data->cancellable =
         _gtk_file_system_get_info (path_bar->priv->file_system,
 				   path_bar->priv->home_file,
-				   "standard::symbolic-icon",
+				   "standard::icon",
 				   set_button_image_get_info_cb,
 				   data);
       break;
 
     case DESKTOP_BUTTON:
@@ -1359,11 +1359,11 @@
 	g_cancellable_cancel (button_data->cancellable);
 
       button_data->cancellable =
         _gtk_file_system_get_info (path_bar->priv->file_system,
 				   path_bar->priv->desktop_file,
-				   "standard::symbolic-icon",
+				   "standard::icon",
 				   set_button_image_get_info_cb,
 				   data);
       break;
     default:
       break;
diff -U 5 -r -Z -B ./org/gtk/gtkplacessidebar.c ./mod/gtk/gtkplacessidebar.c
--- ./org/gtk/gtkplacessidebar.c	2017-07-19 04:55:39.000000000 +0200
+++ ./mod/gtk/gtkplacessidebar.c	2017-07-30 13:48:51.956865078 +0200
@@ -245,29 +245,29 @@
   PROP_POPULATE_ALL,
   NUM_PROPERTIES
 };
 
 /* Names for themed icons */
-#define ICON_NAME_HOME     "user-home-symbolic"
-#define ICON_NAME_DESKTOP  "user-desktop-symbolic"
-#define ICON_NAME_FILESYSTEM     "drive-harddisk-symbolic"
-#define ICON_NAME_EJECT    "media-eject-symbolic"
-#define ICON_NAME_NETWORK  "network-workgroup-symbolic"
-#define ICON_NAME_NETWORK_SERVER "network-server-symbolic"
-#define ICON_NAME_FOLDER_NETWORK "folder-remote-symbolic"
-#define ICON_NAME_OTHER_LOCATIONS "list-add-symbolic"
-
-#define ICON_NAME_FOLDER                "folder-symbolic"
-#define ICON_NAME_FOLDER_DESKTOP  "user-desktop-symbolic"
-#define ICON_NAME_FOLDER_DOCUMENTS      "folder-documents-symbolic"
-#define ICON_NAME_FOLDER_DOWNLOAD       "folder-download-symbolic"
-#define ICON_NAME_FOLDER_MUSIC    "folder-music-symbolic"
-#define ICON_NAME_FOLDER_PICTURES       "folder-pictures-symbolic"
-#define ICON_NAME_FOLDER_PUBLIC_SHARE   "folder-publicshare-symbolic"
-#define ICON_NAME_FOLDER_TEMPLATES      "folder-templates-symbolic"
-#define ICON_NAME_FOLDER_VIDEOS   "folder-videos-symbolic"
-#define ICON_NAME_FOLDER_SAVED_SEARCH   "folder-saved-search-symbolic"
+#define ICON_NAME_HOME     "user-home"
+#define ICON_NAME_DESKTOP  "user-desktop"
+#define ICON_NAME_FILESYSTEM     "drive-harddisk"
+#define ICON_NAME_EJECT    "media-eject"
+#define ICON_NAME_NETWORK  "network-workgroup"
+#define ICON_NAME_NETWORK_SERVER "network-server"
+#define ICON_NAME_FOLDER_NETWORK "folder-remote"
+#define ICON_NAME_OTHER_LOCATIONS "list-add"
+
+#define ICON_NAME_FOLDER                "folder"
+#define ICON_NAME_FOLDER_DESKTOP  "user-desktop"
+#define ICON_NAME_FOLDER_DOCUMENTS      "folder-documents"
+#define ICON_NAME_FOLDER_DOWNLOAD       "folder-download"
+#define ICON_NAME_FOLDER_MUSIC    "folder-music"
+#define ICON_NAME_FOLDER_PICTURES       "folder-pictures"
+#define ICON_NAME_FOLDER_PUBLIC_SHARE   "folder-publicshare"
+#define ICON_NAME_FOLDER_TEMPLATES      "folder-templates"
+#define ICON_NAME_FOLDER_VIDEOS   "folder-videos"
+#define ICON_NAME_FOLDER_SAVED_SEARCH   "folder-saved-search"
 
 static guint places_sidebar_signals [LAST_SIGNAL] = { 0 };
 static GParamSpec *properties[NUM_PROPERTIES] = { NULL, };
 
 static gboolean eject_or_unmount_bookmark  (GtkSidebarRow *row);
@@ -745,11 +745,11 @@
       const gchar *name;
       GIcon *icon;
       int pos = 0;
 
       name = g_file_info_get_display_name (info);
-      icon = g_file_info_get_symbolic_icon (info);
+      icon = g_file_info_get_icon (info);
       uri = g_file_get_uri (file);
       tooltip = g_file_get_parse_name (file);
 
       /* XXX: we could avoid this by using an ancillary closure
        * with the index coming from add_application_shortcuts(),
@@ -786,11 +786,11 @@
 
       if (file_is_shown (sidebar, file))
         continue;
 
       g_file_query_info_async (file,
-                               "standard::display-name,standard::symbolic-icon",
+                               "standard::display-name,standard::icon",
                                G_FILE_QUERY_INFO_NONE,
                                G_PRIORITY_DEFAULT,
                                sidebar->cancellable,
                                on_app_shortcuts_query_complete,
                                sidebar);
@@ -835,11 +835,11 @@
           goto out;
         }
     }
 
   if (info)
-    icon = g_object_ref (g_file_info_get_symbolic_icon (info));
+    icon = g_object_ref (g_file_info_get_icon (info));
   else
     icon = g_themed_icon_new_with_default_fallbacks (clos->is_native ? ICON_NAME_FOLDER : ICON_NAME_FOLDER_NETWORK);
 
   mount_uri = g_file_get_uri (root);
   tooltip = g_file_get_parse_name (root);
@@ -942,11 +942,11 @@
 
   /* add built-in places */
   if (should_show_recent (sidebar))
     {
       mount_uri = "recent:///";
-      icon = g_themed_icon_new_with_default_fallbacks ("document-open-recent-symbolic");
+      icon = g_themed_icon_new_with_default_fallbacks ("document-open-recent");
       add_place (sidebar, PLACES_BUILT_IN,
                  SECTION_COMPUTER,
                  _("Recent"), icon, mount_uri,
                  NULL, NULL, NULL, 0,
                  _("Recent files"));
@@ -1044,11 +1044,11 @@
 
               mount = g_volume_get_mount (volume);
               if (mount != NULL)
                 {
                   /* Show mounted volume in the sidebar */
-                  icon = g_mount_get_symbolic_icon (mount);
+                  icon = g_mount_get_icon (mount);
                   root = g_mount_get_default_location (mount);
                   mount_uri = g_file_get_uri (root);
                   name = g_mount_get_name (mount);
                   tooltip = g_file_get_parse_name (root);
 
@@ -1071,11 +1071,11 @@
                    *
                    * Also, even if automounting is enabled, this gives a visual
                    * cue that the user should remember to yank out the media if
                    * he just unmounted it.
                    */
-                  icon = g_volume_get_symbolic_icon (volume);
+                  icon = g_volume_get_icon (volume);
                   name = g_volume_get_name (volume);
                   tooltip = g_strdup_printf (_("Mount and open “%s”"), name);
 
                   add_place (sidebar, PLACES_MOUNTED_VOLUME,
                              SECTION_MOUNTS,
@@ -1099,11 +1099,11 @@
                *
                * This is mainly for drives like floppies where media detection doesn't
                * work.. but it's also for human beings who like to turn off media detection
                * in the OS to save battery juice.
                */
-              icon = g_drive_get_symbolic_icon (drive);
+              icon = g_drive_get_icon (drive);
               name = g_drive_get_name (drive);
               tooltip = g_strdup_printf (_("Mount and open “%s”"), name);
 
               add_place (sidebar, PLACES_BUILT_IN,
                          SECTION_MOUNTS,
@@ -1147,11 +1147,11 @@
         }
 
       mount = g_volume_get_mount (volume);
       if (mount != NULL)
         {
-          icon = g_mount_get_symbolic_icon (mount);
+          icon = g_mount_get_icon (mount);
           root = g_mount_get_default_location (mount);
           mount_uri = g_file_get_uri (root);
           tooltip = g_file_get_parse_name (root);
           name = g_mount_get_name (mount);
           add_place (sidebar, PLACES_MOUNTED_VOLUME,
@@ -1166,11 +1166,11 @@
           g_free (mount_uri);
         }
       else
         {
           /* see comment above in why we add an icon for an unmounted mountable volume */
-          icon = g_volume_get_symbolic_icon (volume);
+          icon = g_volume_get_icon (volume);
           name = g_volume_get_name (volume);
           add_place (sidebar, PLACES_MOUNTED_VOLUME,
                      SECTION_MOUNTS,
                      name, icon, NULL,
                      NULL, volume, NULL, 0, name);
@@ -1219,11 +1219,11 @@
           network_mounts = g_list_prepend (network_mounts, mount);
           g_object_unref (root);
           continue;
         }
 
-      icon = g_mount_get_symbolic_icon (mount);
+      icon = g_mount_get_icon (mount);
       mount_uri = g_file_get_uri (root);
       name = g_mount_get_name (mount);
       tooltip = g_file_get_parse_name (root);
       add_place (sidebar, PLACES_MOUNTED_VOLUME,
                  SECTION_COMPUTER,
@@ -1258,22 +1258,22 @@
       clos = g_slice_new (BookmarkQueryClosure);
       clos->sidebar = sidebar;
       clos->index = index;
       clos->is_native = is_native;
       g_file_query_info_async (root,
-                               "standard::display-name,standard::symbolic-icon",
+                               "standard::display-name,standard::icon",
                                G_FILE_QUERY_INFO_NONE,
                                G_PRIORITY_DEFAULT,
                                sidebar->cancellable,
                                on_bookmark_query_info_complete,
                                clos);
     }
 
   g_slist_free_full (bookmarks, g_object_unref);
 
   /* Add new bookmark row */
-  new_bookmark_icon = g_themed_icon_new ("bookmark-new-symbolic");
+  new_bookmark_icon = g_themed_icon_new ("bookmark-new");
   sidebar->new_bookmark_row = add_place (sidebar, PLACES_DROP_FEEDBACK,
                                          SECTION_BOOKMARKS,
                                          _("New bookmark"), new_bookmark_icon, NULL,
                                          NULL, NULL, NULL, 0,
                                          _("Add a new bookmark"));
@@ -1306,11 +1306,11 @@
               network_mounts = g_list_prepend (network_mounts, mount);
               continue;
             }
           else
             {
-              icon = g_volume_get_symbolic_icon (volume);
+              icon = g_volume_get_icon (volume);
               name = g_volume_get_name (volume);
               tooltip = g_strdup_printf (_("Mount and open “%s”"), name);
 
               add_place (sidebar, PLACES_MOUNTED_VOLUME,
                          SECTION_MOUNTS,
@@ -1325,11 +1325,11 @@
       network_mounts = g_list_reverse (network_mounts);
       for (l = network_mounts; l != NULL; l = l->next)
         {
           mount = l->data;
           root = g_mount_get_default_location (mount);
-          icon = g_mount_get_symbolic_icon (mount);
+          icon = g_mount_get_icon (mount);
           mount_uri = g_file_get_uri (root);
           name = g_mount_get_name (mount);
           tooltip = g_file_get_parse_name (root);
           add_place (sidebar, PLACES_MOUNTED_VOLUME,
                      SECTION_MOUNTS,
diff -U 5 -r -Z -B ./org/gtk/gtktrashmonitor.c ./mod/gtk/gtktrashmonitor.c
--- ./org/gtk/gtktrashmonitor.c	2017-07-19 04:55:39.000000000 +0200
+++ ./mod/gtk/gtktrashmonitor.c	2017-07-30 13:48:51.960198412 +0200
@@ -50,12 +50,12 @@
 
 G_DEFINE_TYPE (GtkTrashMonitor, _gtk_trash_monitor, G_TYPE_OBJECT)
 
 static GtkTrashMonitor *the_trash_monitor;
 
-#define ICON_NAME_TRASH_EMPTY "user-trash-symbolic"
-#define ICON_NAME_TRASH_FULL  "user-trash-full-symbolic"
+#define ICON_NAME_TRASH_EMPTY "user-trash"
+#define ICON_NAME_TRASH_FULL  "user-trash-full"
 
 static void
 gtk_trash_monitor_dispose (GObject *object)
 {
   GtkTrashMonitor *monitor;
diff -U 5 -r -Z -B ./org/gtk/ui/gtkfilechooserwidget.ui ./mod/gtk/ui/gtkfilechooserwidget.ui
--- ./org/gtk/ui/gtkfilechooserwidget.ui	2017-07-19 04:55:39.000000000 +0200
+++ ./mod/gtk/ui/gtkfilechooserwidget.ui	2017-07-30 13:48:51.960198412 +0200
@@ -71,11 +71,11 @@
                                     <property name="popover">new_folder_popover</property>
                                     <signal name="notify::active" handler="new_folder_popover_active"/>
                                     <child>
                                       <object class="GtkImage">
                                         <property name="visible">1</property>
-                                        <property name="icon-name">folder-new-symbolic</property>
+                                        <property name="icon-name">folder-new</property>
                                         <property name="icon-size">1</property>
                                       </object>
                                     </child>
                                   </object>
                                   <packing>
@@ -291,11 +291,11 @@
                               <class name="dim-label"/>
                             </style>
                             <child>
                               <object class="GtkImage">
                                 <property name="visible">1</property>
-                                <property name="icon-name">edit-find-symbolic</property>
+                                <property name="icon-name">edit-find</property>
                                 <property name="pixel-size">72</property>
                                 <style>
                                   <class name="dim-label"/>
                                 </style>
                               </object>
diff -U 5 -r -Z -B ./org/gtk/ui/gtksidebarrow.ui ./mod/gtk/ui/gtksidebarrow.ui
--- ./org/gtk/ui/gtksidebarrow.ui	2017-07-19 04:55:39.000000000 +0200
+++ ./mod/gtk/ui/gtksidebarrow.ui	2017-07-30 13:48:51.960198412 +0200
@@ -22,12 +22,12 @@
               <object class="GtkBox">
                 <property name="visible">1</property>
                 <child>
                   <object class="GtkImage" id="icon_widget">
                     <property name="visible">1</property>
+                    <property name="margin-right">8</property>
                     <style>
-                      <class name="sidebar-icon"/>
                     </style>
                   </object>
                 </child>
                 <child>
                   <object class="GtkLabel" id="label_widget">
