diff -U 5 -r -Z -B ./org/gtk/gtkscrolledwindow.c ./mod/gtk/gtkscrolledwindow.c
--- ./org/gtk/gtkscrolledwindow.c	2017-09-02 00:17:12.000000000 +0200
+++ ./mod/gtk/gtkscrolledwindow.c	2017-09-13 02:08:37.984051605 +0200
@@ -4408,11 +4408,11 @@
   GtkScrolledWindowPrivate *priv = scrolled_window->priv;
   gboolean use_indicators;
 
   use_indicators = priv->overlay_scrolling;
 
-  if (g_strcmp0 (g_getenv ("GTK_OVERLAY_SCROLLING"), "0") == 0)
+  if (g_strcmp0 (g_getenv ("GTK_OVERLAY_SCROLLING"), "1") != 0)
     use_indicators = FALSE;
 
   if (priv->use_indicators != use_indicators)
     {
       priv->use_indicators = use_indicators;
