Index: src/gtk+-3.24.13/gtk/gtkscrolledwindow.c
===================================================================
--- src.orig/gtk+-3.24.13/gtk/gtkscrolledwindow.c
+++ src/gtk+-3.24.13/gtk/gtkscrolledwindow.c
@@ -4448,7 +4448,7 @@ gtk_scrolled_window_update_use_indicator
 
   use_indicators = overlay_scrolling && priv->overlay_scrolling;
 
-  if (g_strcmp0 (g_getenv ("GTK_OVERLAY_SCROLLING"), "0") == 0)
+  if (g_strcmp0 (g_getenv ("GTK_OVERLAY_SCROLLING"), "1") != 0)
     use_indicators = FALSE;
 
   if (priv->use_indicators != use_indicators)
Index: src/gtk+-3.24.13/gtk/gtksettings.c
===================================================================
--- src.orig/gtk+-3.24.13/gtk/gtksettings.c
+++ src/gtk+-3.24.13/gtk/gtksettings.c
@@ -61,7 +61,7 @@
 #ifdef GDK_WINDOWING_QUARTZ
 #define PRINT_PREVIEW_COMMAND "open -a /Applications/Preview.app %f"
 #else
-#define PRINT_PREVIEW_COMMAND "evince --unlink-tempfile --preview --print-settings %s %f"
+#define PRINT_PREVIEW_COMMAND "atril --unlink-tempfile --preview --print-settings %s %f"
 #endif
 
 /**
Index: src/gtk+-3.24.13/gtk/org.gtk.Settings.FileChooser.gschema.xml
===================================================================
--- src.orig/gtk+-3.24.13/gtk/org.gtk.Settings.FileChooser.gschema.xml
+++ src/gtk+-3.24.13/gtk/org.gtk.Settings.FileChooser.gschema.xml
@@ -133,7 +133,7 @@
       </description>
     </key>
     <key name='startup-mode' enum='org.gtk.Settings.FileChooser.StartupMode'>
-      <default>'recent'</default>
+      <default>'cwd'</default>
       <summary>Startup mode</summary>
       <description>
 	Either "recent" or "cwd"; controls whether the file chooser
