diff -U 10 -r -Z -B ./org/gtk/gtkfilechooserdialog.c ./mod/gtk/gtkfilechooserdialog.c
--- ./org/gtk/gtkfilechooserdialog.c	2017-05-11 21:09:33.000000000 +0200
+++ ./mod/gtk/gtkfilechooserdialog.c	2017-06-18 19:25:59.210159033 +0200
@@ -487,22 +487,20 @@
       GtkWidget *image;
       GtkWidget *header;
 
       button = gtk_toggle_button_new ();
       gtk_widget_set_focus_on_click (button, FALSE);
       gtk_widget_set_valign (button, GTK_ALIGN_CENTER);
       image = gtk_image_new_from_icon_name ("edit-find-symbolic", GTK_ICON_SIZE_MENU);
       gtk_container_add (GTK_CONTAINER (button), image);
       gtk_style_context_add_class (gtk_widget_get_style_context (button), "image-button");
       gtk_style_context_remove_class (gtk_widget_get_style_context (button), "text-button");
-      gtk_widget_show (image);
-      gtk_widget_show (button);
 
       header = gtk_dialog_get_header_bar (GTK_DIALOG (dialog));
       gtk_header_bar_pack_end (GTK_HEADER_BAR (header), button);
 
       g_object_bind_property (button, "active",
                               dialog->priv->widget, "search-mode",
                               G_BINDING_BIDIRECTIONAL);
       g_object_bind_property (dialog->priv->widget, "subtitle",
                               header, "subtitle",
                               G_BINDING_SYNC_CREATE);
diff -U 10 -r -Z -B ./org/gtk/gtkfilechooserwidget.c ./mod/gtk/gtkfilechooserwidget.c
--- ./org/gtk/gtkfilechooserwidget.c	2017-05-11 21:09:33.000000000 +0200
+++ ./mod/gtk/gtkfilechooserwidget.c	2017-06-18 19:27:16.950162886 +0200
@@ -2471,21 +2471,20 @@
 } ColumnMap;
 
 /* Sets the sort column IDs for the file list; needs to be done whenever we
  * change the model on the treeview.
  */
 static void
 file_list_set_sort_column_ids (GtkFileChooserWidget *impl)
 {
   GtkFileChooserWidgetPrivate *priv = impl->priv;
 
-  gtk_tree_view_set_search_column (GTK_TREE_VIEW (priv->browse_files_tree_view), -1);
 
   gtk_tree_view_column_set_sort_column_id (priv->list_name_column, MODEL_COL_NAME);
   gtk_tree_view_column_set_sort_column_id (priv->list_time_column, MODEL_COL_TIME);
   gtk_tree_view_column_set_sort_column_id (priv->list_size_column, MODEL_COL_SIZE);
   gtk_tree_view_column_set_sort_column_id (priv->list_location_column, MODEL_COL_LOCATION_TEXT);
 }
 
 static gboolean
 file_list_query_tooltip_cb (GtkWidget  *widget,
                             gint        x,
@@ -3241,20 +3240,22 @@
 
     case OPERATION_MODE_OTHER_LOCATIONS:
       operation_mode_set_other_locations (impl);
       break;
 
     case OPERATION_MODE_BROWSE:
       operation_mode_set_browse (impl);
       break;
 
     case OPERATION_MODE_SEARCH:
+      priv->operation_mode = old_mode;
+      return;
       operation_mode_set_search (impl);
       break;
 
     case OPERATION_MODE_RECENT:
       operation_mode_set_recent (impl);
       break;
 
     default:
       g_assert_not_reached ();
       return;
@@ -7512,21 +7513,20 @@
 
 static void
 recent_idle_cleanup (gpointer data)
 {
   RecentLoadData *load_data = data;
   GtkFileChooserWidget *impl = load_data->impl;
   GtkFileChooserWidgetPrivate *priv = impl->priv;
 
   gtk_tree_view_set_model (GTK_TREE_VIEW (priv->browse_files_tree_view),
                            GTK_TREE_MODEL (priv->recent_model));
-  gtk_tree_view_set_search_column (GTK_TREE_VIEW (priv->browse_files_tree_view), -1);
 
   gtk_tree_view_column_set_sort_column_id (priv->list_name_column, -1);
   gtk_tree_view_column_set_sort_column_id (priv->list_time_column, -1);
   gtk_tree_view_column_set_sort_column_id (priv->list_size_column, -1);
   gtk_tree_view_column_set_sort_column_id (priv->list_location_column, -1);
 
   update_columns (impl, TRUE, _("Accessed"));
 
   set_busy_cursor (impl, FALSE);
 
diff -U 10 -r -Z -B ./org/gtk/ui/gtkfilechooserwidget.ui ./mod/gtk/ui/gtkfilechooserwidget.ui
--- ./org/gtk/ui/gtkfilechooserwidget.ui	2017-05-11 21:09:33.000000000 +0200
+++ ./mod/gtk/ui/gtkfilechooserwidget.ui	2017-06-18 19:28:50.943500877 +0200
@@ -155,21 +155,22 @@
                             <property name="visible">1</property>
                             <property name="orientation">vertical</property>
                             <child>
                               <object class="GtkScrolledWindow" id="browse_files_swin">
                                 <property name="visible">1</property>
                                 <property name="hscrollbar-policy">never</property>
                                 <child>
                                   <object class="GtkTreeView" id="browse_files_tree_view">
                                     <property name="visible">1</property>
                                     <property name="has-tooltip">1</property>
-                                    <property name="enable-search">0</property>
+                                    <property name="enable-search">1</property>
+                                    <property name="search-column">0</property>
                                     <child internal-child="accessible">
                                       <object class="AtkObject" id="browse_files_tree_view-atkobject">
                                         <property name="AtkObject::accessible-name" translatable="yes">Files</property>
                                       </object>
                                     </child>
                                     <signal name="button-press-event" handler="list_button_press_event_cb" swapped="no"/>
                                     <signal name="drag-data-received" handler="file_list_drag_data_received_cb" swapped="no"/>
                                     <signal name="drag-drop" handler="file_list_drag_drop_cb" swapped="no"/>
                                     <signal name="drag-begin" handler="file_list_drag_begin_cb" swapped="no"/>
                                     <signal name="drag-motion" handler="file_list_drag_motion_cb" swapped="no"/>
