diff -U 10 -r -Z -B ./org/gtk/ui/gtkstatusbar.ui ./mod/gtk/ui/gtkstatusbar.ui
--- ./org/gtk/ui/gtkstatusbar.ui	2017-05-11 21:09:33.000000000 +0200
+++ ./mod/gtk/ui/gtkstatusbar.ui	2017-06-18 20:10:24.643624434 +0200
@@ -1,19 +1,19 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <interface domain="gtk30">
   <!-- interface-requires gtk+ 3.10 -->
   <template class="GtkStatusbar" parent="GtkBox">
     <property name="spacing">2</property>
     <property name="margin-start">10</property>
     <property name="margin-end">10</property>
-    <property name="margin-top">6</property>
-    <property name="margin-bottom">6</property>
+    <property name="margin-top">1</property>
+    <property name="margin-bottom">1</property>
     <child>
       <object class="GtkFrame" id="frame">
         <property name="visible">1</property>
         <property name="shadow-type">none</property>
         <child>
           <object class="GtkBox" id="message_area">
             <property name="visible">1</property>
             <property name="spacing">4</property>
             <property name="margin">4</property>
             <child>
