diff -U 5 -r -Z -B ./org/gtk/gtksettings.c ./mod/gtk/gtksettings.c
--- ./org/gtk/gtksettings.c	2017-07-19 04:55:39.000000000 +0200
+++ ./mod/gtk/gtksettings.c	2017-07-30 13:33:42.580153355 +0200
@@ -59,11 +59,11 @@
 #include "deprecated/gtkrc.h"
 
 #ifdef GDK_WINDOWING_QUARTZ
 #define PRINT_PREVIEW_COMMAND "open -a /Applications/Preview.app %f"
 #else
-#define PRINT_PREVIEW_COMMAND "evince --unlink-tempfile --preview --print-settings %s %f"
+#define PRINT_PREVIEW_COMMAND "atril --unlink-tempfile --preview --print-settings %s %f"
 #endif
 
 /**
  * SECTION:gtksettings
  * @Short_description: Sharing settings between applications
