diff -U 25 -r -Z -B ./org/gtk/gtksettings.c ./mod/gtk/gtksettings.c
--- ./org/gtk/gtksettings.c	2017-05-11 21:09:33.000000000 +0200
+++ ./mod/gtk/gtksettings.c	2017-06-02 03:07:20.300410099 +0200
@@ -39,51 +39,51 @@
 #include <pango/pangofc-fontmap.h>
 #endif
 
 #ifdef GDK_WINDOWING_WAYLAND
 #include "wayland/gdkwayland.h"
 #include <pango/pangofc-fontmap.h>
 #endif
 
 #ifdef GDK_WINDOWING_BROADWAY
 #include "broadway/gdkbroadway.h"
 #endif
 
 #ifdef GDK_WINDOWING_QUARTZ
 #include "quartz/gdkquartz.h"
 #endif
 
 #ifdef GDK_WINDOWING_WIN32
 #include "win32/gdkwin32.h"
 #endif
 
 #include "deprecated/gtkrc.h"
 
 #ifdef GDK_WINDOWING_QUARTZ
 #define PRINT_PREVIEW_COMMAND "open -a /Applications/Preview.app %f"
 #else
-#define PRINT_PREVIEW_COMMAND "evince --unlink-tempfile --preview --print-settings %s %f"
+#define PRINT_PREVIEW_COMMAND "atril --unlink-tempfile --preview --print-settings %s %f"
 #endif
 
 /**
  * SECTION:gtksettings
  * @Short_description: Sharing settings between applications
  * @Title: Settings
  *
  * GtkSettings provide a mechanism to share global settings between
  * applications.
  *
  * On the X window system, this sharing is realized by an
  * [XSettings](http://www.freedesktop.org/wiki/Specifications/xsettings-spec)
  * manager that is usually part of the desktop environment, along with
  * utilities that let the user change these settings. In the absence of
  * an Xsettings manager, GTK+ reads default values for settings from
  * `settings.ini` files in
  * `/etc/gtk-3.0`, `$XDG_CONFIG_DIRS/gtk-3.0`
  * and `$XDG_CONFIG_HOME/gtk-3.0`.
  * These files must be valid key files (see #GKeyFile), and have
  * a section called Settings. Themes can also provide default values
  * for settings by installing a `settings.ini` file
  * next to their `gtk.css` file.
  *
  * Applications can override system-wide settings by setting the property
  * of the GtkSettings object with g_object_set(). This should be restricted
diff -U 25 -r -Z -B ./org/gtk/ui/gtkprintunixdialog.ui ./mod/gtk/ui/gtkprintunixdialog.ui
--- ./org/gtk/ui/gtkprintunixdialog.ui	2017-05-11 21:09:33.000000000 +0200
+++ ./mod/gtk/ui/gtkprintunixdialog.ui	2017-06-02 03:07:20.303743432 +0200
@@ -35,62 +35,62 @@
       <!-- column-name location -->
       <column type="gchararray"/>
       <!-- column-name printer -->
       <column type="GObject"/>
     </columns>
   </object>
   <object class="GtkTreeModelFilter" id="printer_list_filter">
     <property name="child-model">printer_list</property>
   </object>
   <object class="GtkAdjustment" id="scale_spin_adjustment">
     <property name="lower">1</property>
     <property name="upper">1000</property>
     <property name="value">100</property>
     <property name="step-increment">1</property>
   </object>
   <template class="GtkPrintUnixDialog" parent="GtkDialog">
     <property name="can-focus">False</property>
     <property name="type-hint">dialog</property>
     <signal name="notify::page-setup" handler="redraw_page_layout_preview" swapped="no"/>
     <signal name="response" handler="error_dialogs" swapped="no"/>
     <child internal-child="vbox">
       <object class="GtkBox" id="dialog-vbox1">
         <property name="orientation">vertical</property>
         <property name="spacing">2</property>
         <property name="border-width">0</property>
-        <style>
-          <class name="view"/>
-        </style>
         <child>
           <object class="GtkBox" id="box1">
             <property name="visible">1</property>
             <property name="orientation">vertical</property>
             <child>
               <object class="GtkNotebook" id="notebook">
                 <property name="visible">1</property>
                 <property name="can-focus">1</property>
-                <property name="show-border">0</property>
+                <property name="show-border">1</property>
+                <property name="margin-left">12</property>
+                <property name="margin-right">12</property>
+                <property name="margin-top">12</property>
                 <child>
                   <object class="GtkBox" id="general_main_box">
                     <property name="visible">1</property>
                     <property name="border-width">12</property>
                     <property name="orientation">vertical</property>
                     <property name="spacing">18</property>
                     <child>
                       <object class="GtkBox" id="box2">
                         <property name="visible">1</property>
                         <property name="orientation">vertical</property>
                         <property name="spacing">6</property>
                         <child>
                           <object class="GtkScrolledWindow" id="printer_swin">
                             <property name="visible">1</property>
                             <property name="can-focus">1</property>
                             <property name="shadow-type">in</property>
                             <child>
                               <object class="GtkTreeView" id="printer_treeview">
                                 <property name="visible">1</property>
                                 <property name="can-focus">1</property>
                                 <property name="model">printer_list_filter</property>
                                 <signal name="row-activated" handler="emit_ok_response" swapped="no"/>
                                 <child internal-child="selection">
                                   <object class="GtkTreeSelection" id="treeview-selection1">
                                     <property name="mode">browse</property>
