diff -U 10 -r -Z -B ./org/gtk/gtkapplicationwindow.c ./mod/gtk/gtkapplicationwindow.c
--- ./org/gtk/gtkapplicationwindow.c	2017-06-20 21:16:19.000000000 +0200
+++ ./mod/gtk/gtkapplicationwindow.c	2017-07-01 23:27:55.954269814 +0200
@@ -303,21 +303,21 @@
 gtk_application_window_update_shell_shows_app_menu (GtkApplicationWindow *window,
                                                     GtkSettings          *settings)
 {
   gboolean shown_by_shell;
   gboolean shown_by_titlebar;
 
   g_object_get (settings, "gtk-shell-shows-app-menu", &shown_by_shell, NULL);
 
   shown_by_titlebar = _gtk_window_titlebar_shows_app_menu (GTK_WINDOW (window));
 
-  if (shown_by_shell || shown_by_titlebar)
+  if (FALSE && (shown_by_shell || shown_by_titlebar))
     {
       /* the shell shows it, so don't show it locally */
       if (g_menu_model_get_n_items (G_MENU_MODEL (window->priv->app_menu_section)) != 0)
         g_menu_remove (window->priv->app_menu_section, 0);
     }
   else
     {
       /* the shell does not show it, so make sure we show it */
       if (g_menu_model_get_n_items (G_MENU_MODEL (window->priv->app_menu_section)) == 0)
         {
@@ -338,20 +338,21 @@
                   name = g_strdup (app_name);
                 }
               else
                 {
                   /* get the name from .desktop file */
                   name = gtk_application_window_get_app_desktop_name ();
                   if (name == NULL)
                     name = g_strdup (_("Application"));
                 }
 
+              name = g_strconcat("_", _("Application menu"), NULL);
               g_menu_append_submenu (window->priv->app_menu_section, name, app_menu);
               g_free (name);
             }
         }
     }
 }
 
 static void
 gtk_application_window_update_shell_shows_menubar (GtkApplicationWindow *window,
                                                    GtkSettings          *settings)
