Index: gtk+-3.24.30/gtk/gtkwindow.c
===================================================================
--- gtk+-3.24.30.orig/gtk/gtkwindow.c
+++ gtk+-3.24.30/gtk/gtkwindow.c
@@ -4149,15 +4149,10 @@ gtk_window_enable_csd (GtkWindow *window
 {
   GtkWindowPrivate *priv = window->priv;
   GtkWidget *widget = GTK_WIDGET (window);
-  GdkVisual *visual;
 
   /* We need a visual with alpha for client shadows */
   if (priv->use_client_shadow)
     {
-      visual = gdk_screen_get_rgba_visual (gtk_widget_get_screen (widget));
-      if (visual != NULL)
-        gtk_widget_set_visual (widget, visual);
-
       gtk_style_context_add_class (gtk_widget_get_style_context (widget), GTK_STYLE_CLASS_CSD);
     }
   else
@@ -7371,10 +7366,18 @@ gtk_window_realize (GtkWidget *widget)
   GtkWindowPrivate *priv;
   gint i;
   GList *link;
+  GdkScreen *screen;
+  GdkVisual *visual;
 
   window = GTK_WINDOW (widget);
   priv = window->priv;
 
+  screen = gtk_widget_get_screen (widget);
+  visual = gdk_screen_get_rgba_visual (screen);
+
+  if (gdk_screen_is_composited (screen) && visual != NULL)
+    gtk_widget_set_visual (widget, visual);
+
   if (!priv->client_decorated && gtk_window_should_use_csd (window))
     create_decoration (widget);
 
